xbstrap/base.py

```diff
@@ -1704,13 +1713,16 @@ def fetch_src(cfg, src):
 					+ ':' + 'refs/remotes/origin/' + source['branch']])
 		subprocess.check_call(args, cwd=src.source_dir)
 	elif 'hg' in source:
+		if shutil.which('hg') is None:
+			raise GenericException("mercurial (hg) not found; please install it and retry")
```

Comment:
If we do use `which`, we should also use the result of `shutil.which` to invoke the subprocess. (Otherwise, there is a TOCTOU condition).